update params structure for OSM hosted iD

answerquest opened the pull request:
update params structure for the iD editor hosted at openstreetmap.org/edit : have to keep the params after a # instead of ?

Solves https://github.com/openstreetmap/iD/issues/9093  and https://github.com/openstreetmap/openstreetmap-website/issues/3540

answerquest committed:
update params structure for OSM iD

update params structure for the iD editor hosted at openstreetmap.org/edit : have to keep the params after a # instead of ?

mmd-osm commented on API.md:
```diff
@@ -67,7 +67,8 @@ of iD (e.g. `https://ideditor-release.netlify.app`), the following parameters ar
 Port (e.g. `http://www.openstreetmap.org/edit?editor=id`), the following parameters
-are available as **regular URL query parameters**:
+are available as **URL hash-query parameters, to put after a # instead of a ?**.  
+_Example:_ `https://www.openstreetmap.org/edit#gpx=https://tasks.hotosm.org/project/592/task/16.gpx`
```
Unfortunately, the link to that GPX file doesn't seem to work here.

answerquest replied:
Yes, but same link is mentioned in the same page above also. I assumed keeping the same here will give a consistent logical flow to help someone reading the documentation to follow properly. I do have valid GPX links but it'll end up exposing my server to more bots etc. Let me try hosting one in a safe place.

answerquest replied:
ok how about this: https://www.openstreetmap.org/edit#gpx=https://gist.githubusercontent.com/answerquest/9445352b60ca5b44714675eae00f243a/raw/56a6343a29223318f4a697bfd16cbb2c3b8155ad/sample_boundary.gpx

answerquest replied:
updated. pls tell if you need me to change the gpx link in top portion of the page too in the standalone iD section. I'll prefer to leave that be.

mmd-osm replied:
Example works fine now, thanks!

answerquest committed:
working osm editor with-gpx-hash-param link

tyrasd committed:
update list of query/hash params on osm.org 

Some hash parameters are not supported on osm.org, while there are some query parameters which work slightly differently when used on osm.org 
See https://github.com/openstreetmap/openstreetmap-website/blob/66656e1d1d58e6073efd453007d7452920531b37/app/assets/javascripts/edit/id.js.erb

This should accurately the currently state of the iD integration in the openstreetmap-website (aka as the rails port).

tyrasd commented:
I added some clarifications to this PR in 93713a41adb745ab2b25309d752258eba1d6057c: In fact, some of the listed parameters are actually query parameters (e.g. `lat`,`lon`,`zoom`,`node`,`way`,`relation`, `locale`).

Slightly related: There exist a small bug in the openstreetmap website which in some situations caused "all" query parameters to be interpreted like hash parameters. https://github.com/openstreetmap/openstreetmap-website/pull/3571 has a fix for that.
